### PR TITLE
[bitnami/moodle] fix MOODLE_PASSWORD with existingSecrets

### LINKED ISSUE #23214: [bitnami/moodle] existing secrets not working
### Name and Version

bitnami/moodle:20.0.1

### What architecture are you using?

amd64

### What steps will reproduce the bug?
```
  podAnnotations: {}
  allowEmptyPassword: false
  # -- Skip Moodle™ installation wizard. Useful for migrations and restoring from SQL dump
  moodleSkipInstall: false
  moodleUsername: admin
  existingSecret: "moodle-secrets"
  ingress:
    enabled: false

  mariadb:
    enabled: false
  externalDatabase:
    user: usermoodle
    # password: uCUCeyc783ew6W3xQZmB873
    database: moodle
    type: mariadb
    port: 3306
    existingSecret: "moodle-secrets"

```
### Are you using any custom parameters or values?

all the existingSecret sequence won't work , because they include `common.names.fullname`
instead of something similar to this https://github.com/bitnami/charts/blob/67681dcb1608db89da63d6797547f0bbd98d467a/bitnami/wordpress/templates/deployment.yaml#L153

### What is the expected behavior?

can map correctly the existing secret upon declaration 

### What do you see instead?

instead i see `.Release.Name-moodle` 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by DrummyFloyd**
### Description of the change

[bitnami/moodle] : fix the usage of `existingSecret` like other helm chart 

### Benefits

bitnami standard
### Possible drawbacks

.

### Applicable issues

- fixes #23214

### Additional information

.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/moodle/templates/_helpers.tpl [hunk 1]
```diff
@@ -142,3 +142,14 @@ Return the MariaDB Secret Name
     {{- printf "%s-%s" (include "common.names.fullname" .) "externaldb" -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Return the Moodle Secret Name
+*/}}
+{{- define "moodle.secretName" -}}
```
Comment (fmulero) [inline bitnami/moodle/templates/_helpers.tpl:149]: This helper already exists, please take a look to the line 70: https://github.com/bitnami/charts/blob/ee19ef71e5f0522013e98c3bae7ed47ce0362920/bitnami/moodle/templates/_helpers.tpl#L67-L72

**Review by fmulero — CHANGES_REQUESTED** [review]
Thanks @DrummyFloyd for your contribution.

Please take a look to my comment. The new helper is not needed.

File: bitnami/moodle/templates/_helpers.tpl (hunk 1 above)
Comment (DrummyFloyd) [reply bitnami/moodle/templates/_helpers.tpl:149]: Une right , did not see , will remove my patch on helper.txt

Comment (DrummyFloyd) [reply bitnami/moodle/templates/_helpers.tpl]: i remove the old function and use the standard'one

**Review by DrummyFloyd — COMMENTED** [review]
is this ok for you ?

File: bitnami/moodle/templates/_helpers.tpl [hunk 2]
```diff
@@ -68,7 +68,11 @@ Return  the proper Storage Class
 Moodle&trade; credential secret name
 */}}
 {{- define "moodle.secretName" -}}
-{{- coalesce .Values.existingSecret (include "common.names.fullname" .) -}}
+{{- if .Values.existingSecret }}
+    {{- printf "%s" .Values.existingSecret -}}
+{{- else -}}
+    {{- printf "%s" (include "common.names.fullname" .) -}}
+{{- end -}}
```
Comment (fmulero) [inline bitnami/moodle/templates/_helpers.tpl:75]: I don't see right now any reason to change this piece of code. `coalesce` do exactly [the same](https://helm.sh/docs/chart_template_guide/function_list/#coalesce). Please amend me if I am wrong

**Review by fmulero — COMMENTED** [review]
Sorry @DrummyFloyd  to bother you again, but there is a change I don't fully understand. Could you give it a look?

File: bitnami/moodle/templates/_helpers.tpl (hunk 2 above)
Comment (DrummyFloyd) [reply bitnami/moodle/templates/_helpers.tpl]: your right my bad
thank you for your review.

**Review by fmulero — APPROVED** [review]
LGTM

**Comment by DrummyFloyd** [comment]
thank you again, 
correct me if i'm wrong, but it should be already deploy on 
https://artifacthub.io/packages/helm/bitnami/moodle#views ?

**Comment by carrodher** [comment]
This version (20.2.3) was released in DockerHub which is the place where Bitnami Helm charts are released, see https://hub.docker.com/r/bitnamicharts/moodle/tags

**Comment by DrummyFloyd** [comment]
> This version (20.2.3) was released in DockerHub which is the place where Bitnami Helm charts are released, see https://hub.docker.com/r/bitnamicharts/moodle/tags

how the PR contains the 20.2.3 version and on docker hub the version seems to be release 6 days ago ? (20.2.3)

**Comment by fmulero** [comment]
My failure. It seems version 20.2.3 was released before merging these changes. I didn't notice it and  It didn't cause any conflicts because the version in your PR and the one in the main branch were the same.  To amend this situation I've just bumped the chart version to release your changes. Sorry about the inconveniences.

**Comment by DrummyFloyd** [comment]
shit happnd, don't worry ^^
thank you @fmulero

**Pull request merged**